crates/bevy_ecs/src/observer/mod.rs

```diff
@@ -122,6 +123,20 @@ impl<'w, E: Debug, B: Bundle> Debug for Trigger<'w, E, B> {
     }
 }
 
+impl<'w, E, B: Bundle> Deref for Trigger<'w, E, B> {
+    type Target = E;
+
+    fn deref(&self) -> &Self::Target {
+        self.event
+    }
+}
+
+impl<'w, E, B: Bundle> DerefMut for Trigger<'w, E, B> {
+    fn deref_mut(&mut self) -> &mut Self::Target {
+        self.event
+    }
+}
```

Comment:
Was there a particular reason this was added in this PR? I don't disagree with the implementation, but it feels tangential to the goal here.

Reply:
I added `Deref`/`DerefMut` impls for all `SystemInput`s for future generic usage of abstracting over mutable or readonly access to a type in the input: `where I: SystemInput + DerefMut<Target = String>`, for example. Since `Trigger` is now a `SystemInput`, I added it for equality. If it's a controversial change I'm fine with removing it, but thought I might as well since we're mucking around with observers here.

Reply:
No that seems perfectly reasonable, thanks for clarifying! Just wanted to get the reasoning written down in-case anyone else had the same thought.